### PR TITLE
Move NET publish to release action. Trigger from release. Approval step

### PR DESCRIPTION
**Pull request by tmarkovski**

### CODE REVIEW

File: .github/workflows/release-dotnet.yml [hunk 1]
```diff
@@ -2,25 +2,62 @@ name: Release .NET SDK
...
   workflow_dispatch:
-  release:
-    types: [published]
+  # release:
+  #   types: [published]
+  workflow_run:
+    workflows: ["Release Platform Libraries"]
+    branches: [main]
+    types:
+      - completed
```
Comment (fundthmcalculus) [inline .github/workflows/release-dotnet.yml:11]: oooh, that's cool! Can we add a manual trigger as well for parity with other actions?

File: .github/workflows/release-dotnet.yml [hunk 2]
```diff
@@ -2,25 +2,62 @@ name: Release .NET SDK
...
     runs-on: windows-latest
     steps:
       - name: Download workflow artifact
-        uses: dawidd6/action-download-artifact@v2.14.0
+        uses: actions/download-artifact@v2
         with:
-          workflow: "build-dotnet.yml"
-          path: ./nuget
-          branch: main
+          name: nuget
```
Comment (fundthmcalculus) [inline .github/workflows/release-dotnet.yml:60]: Nice! I'll make a PR that does similar things for the other languages.

**Review by fundthmcalculus — APPROVED** [review]
See comments before merge, please.

File: .github/workflows/release-dotnet.yml (hunk 2 above)
Comment (tmarkovski) [reply .github/workflows/release-dotnet.yml:60]: Let's do few iterations of this process first on the .NET stream before we apply it. I've some ideas I wanna run by you

File: .github/workflows/release-dotnet.yml (hunk 1 above)
Comment (tmarkovski) [reply .github/workflows/release-dotnet.yml:11]: Yeah, there's `workflow_dispatch` trigger above.

**Pull request merged**